test/native/util.test.js

```diff
@@ -0,0 +1,221 @@
+import { tasked, destroyer } from '../../native/utils';
+import { describe, it } from 'mocha';
+import assume from 'assume';
+
+describe('(native) util', function () {
+  describe('#tasked', function () {
+    it('is a function', function () {
+      assume(tasked).is.a('function');
+    });
+
+    it('returns an object', function () {
+      const api = tasked(() => {});
+
+      assume(api).is.a('object');
+      assume(api).is.length(2);
+      assume(api.run).is.a('asyncfunction');
+      assume(api.done).is.a('asyncfunction');
+    });
+
+    describe('#run', function () {
+      it('executes the task', async function () {
+        const { run } = tasked(() => { throw new Error('fail'); });
+        let called = false;
+
+        await run(() => {
+          called = true;
+        });
+
+        assume(called).true();
+      });
+
+      it('receives the done as argument', async function () {
+        const { run, done } = tasked(() => { throw new Error('fail'); });
+        let what;
+
+        await run((cb) => {
+          what = cb;
+        });
+
+        assume(what).equals(done);
```

Comment:
What's the use-case for this? I don't see any usages of it right now, anytime you want done you're just getting it from `tasked` directly.  It's also an async method so it should either get returned by the function passed to `run` when called, or awaited. so that an error handler can listen to failures. This part feels weird to me.

Reply:
Yes, it might look weird in this test, because it does return the done function as well as the runner. But when assume that the tasks that are ran, and passed in from plugins, that do not have access to the task runner it self, it does make sense, as they will be able to prevent the execution of the rest of the tasks.

Reply:
Should there be a test that uses it like a plugin will?